admin/admin.go

```diff
@@ -0,0 +1,179 @@
+package admin
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/vocdoni/blind-csp/model"
+	"go.vocdoni.io/dvote/httprouter"
+	"go.vocdoni.io/dvote/httprouter/bearerstdapi"
+	"go.vocdoni.io/dvote/log"
+)
+
+const bearerTokenReqAmount = 100000000
+
+type Admin struct {
+	router             *httprouter.HTTProuter
+	api                *bearerstdapi.BearerStandardAPI
+	storage            *model.MongoStorage
+	electionController *ElectionController
+	userController     *UserController
+}
+
+func NewAdmin() *Admin {
+	admin := new(Admin)
+
+	// Storage
+	admin.storage = &model.MongoStorage{}
+	if err := admin.storage.Init(); err != nil {
+		log.Fatal(err)
+	}
+
+	// Models & Controllers
+	electionStore := model.NewElectionStore(admin.storage)
+	admin.electionController = NewElectionController(electionStore)
+
+	userStore := model.NewUserStore(admin.storage)
+	admin.userController = NewUserController(userStore)
+
+	return admin
+}
+
+func (admin *Admin) ServeAPI(r *httprouter.HTTProuter, baseRoute string) error {
+	if !strings.HasPrefix(baseRoute, "/") {
+		return fmt.Errorf("invalid base route %q, it must start with /", baseRoute)
+	}
+
+	// Remove trailing slash
+	if len(baseRoute) > 1 {
+		baseRoute = strings.TrimSuffix(baseRoute, "/")
+	}
+	if r == nil {
+		return fmt.Errorf("router is nil")
+	}
+
+	var authToken []string
+	if len(authToken) == 0 {
+		authToken = strings.Split(os.Getenv("ADMINAPI_AUTHTOKEN"), ",")
+	}
+
+	// Initialize API
+	admin.router = r
+	var err error
+	admin.api, err = bearerstdapi.NewBearerStandardAPI(admin.router, baseRoute)
+	if err != nil {
+		return err
+	}
+
+	// Create the Bearer API attached to the router
+	admin.api.EnableVerboseAuthLog()
+
+	// Set bearer authentication
+	if len(authToken) == 0 || authToken[0] == "" {
+		authToken = []string{uuid.New().String()}
+	}
+
+	for _, at := range authToken {
+		admin.api.AddAuthToken(at, bearerTokenReqAmount)
+	}
+	log.Infof("using bearer authentication token %s", authToken)
```

Comment:
log.Infow, log.Debugw, log.Warnw and log.Errorw are the new methods for our log package that should be used when writting new code. 

Syntax is: log.Infow("message", key,1 value1, key2, value2, ...)

`log.Infow("using bearer authentication", "token", authToken)`